erts/preloaded/src/prim_inet.erl

```diff
@@ -520,13 +520,34 @@ sendfile(S, FileHandle, Offset, Length)
 sendfile(S, FileHandle, Offset, Length) ->
     case erlang:port_info(S, connected) of
         {connected, Pid} when Pid =:= self() ->
-            sendfile_1(S, FileHandle, Offset, Length);
+            Uncork = sendfile_maybe_cork(S),
+            Result = sendfile_1(S, FileHandle, Offset, Length),
+            sendfile_maybe_uncork(S, Uncork),
```

Comment:
Dialyzer complains about unmatched return values here and on line 538. I believe it should be safe to suppress these warnings (`_ = ...`) and have done so for the nightly builds.

You can check this yourself with `scripts/run-dialyzer` after rebuilding preloaded modules.

Reply:
Updated with a fix, let me know if it still complains about anything that I didn't notice.